ReadTheDocs updates for GCHP 14.5.0

yantosca opened the pull request:
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Describe the update

This PR seeks to merge documentation updates for GCHP 14.5.0 into the dev/14.5.0 branch.  We have made the following general updates:

- Moved documentation for `geoschem_config.yml`, `HEMCO_Diagn.rc`, `HEMCO_Config.rc`, `species_database.yml`, and the aerosol/chemistry/photolysis configuration files to the https://github.com/geoschem/geos-chem-shared-docs repo.
- Now point to relocated documentation files in the geos-chem-shared-docs repo
- Added usage differences between GCHP and GCC for the various documentation files
- Updated formatting to improve the page rendering and usability (i.e. using `.. option::` tags, etc)

### Related Github Issue
- https://github.com/geoschem/GCClassic/pull/68
- https://github.com/geoschem/HEMCO/pull/292

### NOTES
For now I will leave this as a draft PR as we will might have more minor updates to push.  But feel free to make comments & request changes on this PR now.  Tagging @msulprizio @lizziel @yidant

yantosca committed:
GEOS-Chem submod update: Merge PR #2359 (Per-species KPP error tol.)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

4288f0d1d Merge PR #2359 (Set KPP dummy species to large ATOL value)

This PR adds structural updates to allow per-species setting of the
KPP absolute & relative tolerances.  This now allows us to assign a
large absolute tolerance (1e25) to KPP dummy species, which excludes
them from the error norm computations.  This results in a faster
simulation at the cost of numerical noise.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge hotfix from PR #2363 atop PR #2359

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

e61a51064 Merge GCHP hotfix from PR #2363 on top of PR #2359

This PR implements a hotfix for an issue that had caused GCHP
cloud benchmarks to fail.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2315 (complexSOA bug fixes)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

ef2ea60e7 Merge PR #2315 (Fix bugs in PDER, OCPISOA, and TotalOC)

This PR fixes several minor bugs in the complex SOA scheme.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Merge GCHP 14.4.2 back into the 14.5.0 development stream

This merge brings the GCHP 14.4.2 release into the GCHP
14.5.0 development stream.  14.4.2 is a zero-diff update.  The
CHANGELOG.md has been updated accordingly.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2398 (Uncomment cations for HETP)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

e29a4c928 Merge PR #2398 (Pass non-zero Ca2, Mg, and K cations to HETP)

PR #2398 uncomments setting Ca2+, K+, and Mg+ cation values passed
to HETP. Previously these values were zero due to an issue in ISORROPIA.
With the update to HETP we can enable these cations.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2352 (ALK4/R4N2 fixes from J. Brewer)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

0a392d465 Merge PR #2352 (Fixes for ALK4 and R4N2 chemistry from J. Brewer)

R4N2 was a common product of alkane, isoprene, and monoterpene oxidation
and produced organonitrate SOA but we don’t want ALK4 to produce
organonitrate SOA as discusses in #1625. A fix from Jared Brewer is
included in this PR.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: PR #2352 post-merge fix (rebuild KPP files)

This commit informs the GCClassic superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

acd335486 PR #2352 post-merge fix: Rebuild fullchem solver files

This PR restores the updated KPP solver files for PR #2352 (ALK4 and
R4N2 chemistry updates, by @JFBrewer).

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Merge GEOS-Chem 14.4.3 back into the 14.5.0 development branch

This merge brings all of the updates from the GEOS-Chem 14.4.3
"no-diff-to-benchmark" release back into the dev/14.5.0 branch.
We will test this merge with its own alpha benchmark.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: PR #2425 post-merge fix (CHANGELOG.md update_

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

8f0a16767 PR #2425 post-merge fix: Move 14.4.3 entry to top of CHANGELOG.md

This PR moves a CHANGELOG.md entry for 14.4.3 to its proper place.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem & HEMCO updates: Merge PR #2318, PR #284, PR #285

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

586260f10 PR #2318 post-merge fix: Use less drydep fields for the alldiags test
f240c993f PR #2318 post-merge fix: Remove duplicate EmisHCOOH_Total entry
53f9b2e82 PR #2318 post-merge fix: Add missing species to geoschem_config.yml
dd7ba9a9c Merge PR #2318 (Chemistry for RCOOH, ANs, PNs)

and the geoschem/HEMCO repository:

d0d5fca PR #285 post-merge fix: Update CHANGELOG.md
1c6356d Merge PR #285 (Add em. factors for ALK6, C4H6, EBZ, STYR, TMB)
253b0fc Merge HEMCO 3.9.3 back into the 3.10.0 development stream
f2b6ae4 Merge PR #284 (Fix inconsistent pressure units in HEMCO standalone)

These PRs implement new chemistry (from @ktravis213), emissions (from
@kelvinhb), and a bug fix for the HEMCO standalone.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2428 (vdiff) and PR #2144 (drydep)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

3073aff2e Merge PR #2428 (Fix mass non-conservation in VDIFF)
c7332ed7a Merge PR #2144 (Restore drydep output in fullchem benchmarks)

This PR does the following:

1. Adds the DryDep collection to all fullchem_benchmark runs
2. Implements a mass conservation fix in VDIFF PBL mixing by
   @nicholasbalasus.  See issue #2409 for details/

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
RTD update: Update compiler to 12.2.0; Update page formatting

docs/source/geos-chem-shared-docs
- Changed compiler from gcc@10.2.0 to gcc@12.2.0 in guides

docs/source/user-configuration-files.rst
docs/source/user-guide/rundir-init.rst
docs/source/user-guide/running.rst
- Updated formatting and fixed issues flagged by Sphinx

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2436, #2439, #2371 (PPN Phot etc.)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

602b660c1 PR #2436 post-merge update: Fix typo in ExtData.rc.fullchem
d34c567a4 Merge PR #2436 (Update GFED4 climatology for 2010-2023)
d4c47babf Merge PR #2439 (Update run scripts and env files for Discover)
43d0e9083 Merge PR #2371 (Adding PPN photolysis)

This implements photolysis for the PPN species.  Also, two other
no-diff-to-benchmark updates have been added, namely updating run scripts
for NASA discover and updating the GFED4 climatology file.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2424, PR #2426, Cloud-J PR #24

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

1951ea96e Declare PPN as photolyzing in the species database
40a078514 Merge PR #2424 (Support for Cloud-J_v8.0.0)
647c9f579 Merge PR #2426 (Store aerosol-specific optics files in new directory)

and to the geoschem/Cloud-J repository:

291a161 Update Cloud-J standalone settings to use UV absorption by H2O by default
6a027b1 Merge pull request #24 from geoschem/feature/cloudj_v8

These PRs move aerosol optics configuration files to a separate folder
(out of the CHEM_INPUTS/FAST-JX folder structure) and update Cloud-J
to version 8.0.0.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Update jinja2 to 3.1.4 in docs/requirements.txt

docs/requirements.txt
- Use jinja 3.1.4 as there is a dependabot security warning for
  the previous version 3.1.3.

CHANGELOG.md
- Updated accordingly

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Cloud-J submod update: Merge PR #26 (Define max J-values rather than exact)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

6bf4a36 Merge pull request #26 from geoschem/feature/define_max_rather_than_exact_num_jvalues

This PR fixes an issue where Cloud-J will stop reading entries from
FJX_j2j.dat before the "9999" end-of-file marker.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem submod update: Merge PR #2452 (Reorder photo rxns for PNs, ANs)

This commit informs the GCHPsuperproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

a5fba9c88 Merge PR #2452 (Reorder photo rxns for PNs and ANs)

This PR reorders the photo rxns of the PN and AN species added by
@ktravis213 in PR #2318, so that they now follow PPN in the
FJX_j2j.dat configuration file for Cloud-J.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2457 (Update Is_Photolysis tags)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

cd0366015 Merge PR #2457 (Update Is_Photolysis tags in species_database.yml)

This PR ensures that photolyzing species (as listed in the KPP
fullchem.eqn and FJX_j2j.dat files) have "Is_Photolysis.true" tags
in the species_database.yml file.  Non-photolyzing species have
had these tags removed.  Also the JValues collection in the GCHP
HISTORY.rc.fullchem file has been updated.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem & HEMCO updates: Merge PR #2462, PR #2430, PR #287

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

c0aeff427 Merge PR #2462 (Bug fix: ALK6 in HEMCO_Config.rc should be ALK7)
5c31c3470 Merge PR #2430 (Updated Y. Wang soil NOx scheme as an option)

and the geos-chem/HEMCO repository

c6a9d46 Merge PR #287 (Updated Y. Wang soil NOx scheme as an option)

This PR does the following:

  1. Implements the U. Iowa soil NOx scheme as an option.
     This is currently switched off by default.

  2. Adds a bug fix in HEMCO configuration files: ALK6 should
     be called ALK7.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submodupdate: Merge PR #2464 (remove zero J-value diagnostics)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submod update: Merge PR #1467 (Fix dry mass fraction of seasalt in aerosol thermodynamics)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
Bugfix: Restore stale workflow functionality

The stale workflows are currently failing with the error:

    Invalid workflow file: .github/workflows/stale.yml#L26
    The workflow is not valid. .github/workflows/stale.yml (Line: 26, Col: 11): A sequence was not expected

To fix this, the list of exempt-issues-labels in .github/workflows/stale.yml
must be a  comma separated list. A single quote is placed around the list to
allow for use of colons in label names.

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submod update: Merge PR #2477 (Fix for double counting CH4 ag burning emissions)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submodule update: Merge PR #2473 (seasonality of CH4 reservoir emissions)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem and HEMCO submodule updates: Add fixes for stale issue Github workflow

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
Merge PR #444 containing a fix for the stale issue Github workflow

msulprizio committed:
GEOS-Chem submodule update: Merge PR #2489 (fix for coarse-mode seasalt Ca, Mg, K)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submodule update: Merge PR #2490 (clean up CHANGELOG.md)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
Cloud-J submodule update for 8.0.1 release

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

yantosca committed:
Update GCHP configuration file documentation

- Now point to geoschem_config.yml, HEMCO_Config.rc, and HEMCO_Diagn.rc,
  species_database.yml, and aerosol/chem/photolysis configuration files
  in the documentation in geos-chem-shared-docs/doc folder
- Updated the configuration file overview page accordingly
- Now use ".. option::" tags for easier searching of items
- Reformatted text using ReST best practices
- Trimmed trailing whitespace

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
RTD Update: Fix typo in cap_restart doc page

docs/source/user-guide/config-files/cap_restart
- Fixed typo ":ref:`set-common-run-settings.sh`" ->
  ":ref:`set-common-run-settings-sh`".

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Update text in cap_restart for clarity

docs/source/user-guide/config-files/cap_restart
- Updated the text so that it flows better and is less awkward

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Minor updates in config file doc; Update geos-chem-shared-docs

docs/source/geos-chem-shared-docs
- Added GCHP vs. GCC usage differences section in HEMCO_Diagn.rc doc

docs/source/user-guide/configuration-files.rst
- Now use reference links to the setCommonRunSettings.sh doc in the
  NOTE at the top of the page
- Now split the configuration files into "Commonly-updated configuration
  files" and "Less commonly-updated configuration files" to make it
  clear that some config files don't normally need to be updated
  except under certain circumstances.
- Trimmed whitespace

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
geos-chem-shared-docs update: Fixed typo in geoschem_config.yml doc

docs/source/geos-chem-shared-docs:
- Fixed typo ":ref:`ExtData.rc`" -> ":file:`ExtData.rc`"

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
ReadTheDocs update: Update formatting & links in more files

docs/source/user-guide/compiling.rst
docs/source/user-guide/getting-input-data.rst
docs/source/user-guide/running.rst
- Updated formatting using ReST best practices
- Link to pages with references where expedient
- Trim trailing whitespace

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

msulprizio committed:
GEOS-Chem submodule update: Merge PR #2493 (rename ALK7 to ALK6)

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

msulprizio committed:
GEOS-Chem and MAPL submodule updates: Merge fixes for GCHP carbon simulation

- geos-chem: Merge PR #2492 fixes for GCHP carbon simulation
- MAPL: Merge PR #37 conatining fix to vertically flip imports with dimensionless pressure proxy lev coordinates

Signed-off-by: Melissa Sulprizio <mpayer@seas.harvard.edu>

yantosca committed:
geos-chem-shared-docs update: GC input data; GCHP History guide

This commit informs the GCHP superproject about the following
commits that were pushed to the geos-chem-shared-docs repository:

65b2a35 Merge "docs/update_guides_for_gchp" into "main"
7f0cfbb Merge PR #3 (Add doc/tutorial for GEOS-Chem input data on AWS)

These are documentation updates for the 14.5.0 release.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

lizziel committed:
Specify upper limit on gfortran supported in GCHP

Current MAPL does not reliably build with GCC 13.

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

yantosca committed:
RTD update: Fix broken links/refs; Improve text formatting etc.

docs/source/geos-chem-shared-docs
- Submodule update to cd0f4ad, w/ recent doc updates for 14.5.0

docs/source/getting-started/quick-start.rst
- Add "Important" box for GEOS-FP convection
- Now use consistent $ prompt in examples

docs/source/index.rst
- Update toctree link to the GEOS-Chem Input Data on AWS, which
  was renamed to geos-chem-shared-docs/doc/gcid-portal-overview.rst

docs/HEMCO_Config.rc
docs/HEMCO_Diagn.rc
- Removed; Documentation for these is in geos-chem-shared-docs now

docs/source/user-guide/config-files/HISTORY.rc
- Fixed incorrect option name: "ref_date:" -> "ref_date"

docs/source/user-guide/config-files/setCommonRunSettings.sh
- Fixed incorrect option name: "extdata" -> "extdata-rc"

docs/source/user-guide/debugging.rst
- Added |br| break definition
- Fixed links to the debug-guide and error-guide in geos-chem-shared-docs
- Updated code formatting to fit w/in 80 cols (mostly)
- Trimmed trailing whitespace

docs/source/user-guide/getting-input-data.rst
- Add text and link to new GEOS-Chem Input Data portal doc
- Updated code formatting to fit w/in 80 cols (mostly)
- Trimmed trailing whitespace

docs/source/user-guide/rundir-config.rst
- Removed "-----" breaks
- Updated code formatting to fit w/in 80 cols (mostly)
- Now use proper heading section delimiters
- Trimmed trailing whitespace

docs/source/user-guide/rundir-init.rst
- Add "Important" box for GEOS-FP convection

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
geos-chem-shared-docs update: Link to dry-run doc from GC Input Data doc

docs/source/geos-chem-shared-docs
- RTD update: Link to dy-run instructions from GC Input Data doc

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Submod update: Merge GC PR #2555 & HEMCO PR #295 (14.5.0 dry run)

This commit informs the GCClassic superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

a8bb552d1 Merge PR #2555 (Dry run updates for GEOS-Chem Classic 14.5.0)
e9d951f Merge PR #295 (Add HEMCO-standalone dry run scripts)

This PR updates the subroutine hashes to GEOS-Chem and HEMCO submodules,
which now have updates for the GEOS-Chem 14.5.0 and HEMCO 3.10.0
dry run simulations.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

msulprizio commented on docs/source/user-guide/compiling.rst:
```diff
@@ -54,121 +79,158 @@ argument:
-   For example, if you got an error saying "Could not find ESMF", and ESMF is installed
-   to :file:`/software/ESMF`, you would do
+   For example, if you got an error saying "Could not find ESMF", and
+   ESMF is installedto :file:`/software/ESMF`, you would do
```
Minor typo here. Need a space between installed and to.

lizziel replied:
Fixed.

msulprizio commented on docs/source/user-guide/config-files/ExtData_rc.rst:
```diff
@@ -1,57 +1,164 @@
+for GCHP. Explanatory information about the file is located at the top
+of the configuration file in all run directories. The file format is
+the same as that used in the GEOS model, and GMAO/NASA documentation
+for it can be found at the ExtData component page on the GEOS-5
```
Should we add a link here?

lizziel replied:
Link added.

lizziel committed:
Update changelog, add ExtData docs link, and minor fixes/edits

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Fix typos in links

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

yantosca committed:
Updated ReadTheDocs documentation for the 14.5.0 release

CHANGELOG.md
CMakeLists.txt
docs/source/conf.py
src/GEOS-Chem/CHANGELOG.md
src/GEOS-Chem/KPP/fullchem/CHANGELOG_fullchem.md
- Updated version numbers from 14.4.3 to 14.5.0

docs/requirements.txt
- Update Python package jinja2 needed for ReadTheDocs from
  3.1.3 to 3.1.4 (fixes a security issue)

docs/source/getting-started/*.rst
docs/source/user-guide/*.rst
docs/source/geos-chem-shared-docs
- Updated documentation for 14.5.0.  Includes:
  - New documentation for the GEOS-Chem Input Data portal on AWS
  - Corresponding updates to the dry-run simulation docs
  - Moved doc common to GCClassic, GCHP, HEMCO to geos-chem-shared-docs
  - Added warnings in doc about GEOS-FP convection update on 6/1/20
  - Updated links and references
  - Text formatting updates (indentation, trimmed whitespace, etc.)

src/GCHP_GridComp/GEOSChem_GridComp/geos-chem
- Submodule update (updated version numbers to 14.5.0)

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GCHP 14.5.0 release

Updated version numbers in:
- CHANGELOG.md
- CMakeLists.txt
- docs/source/conf.py

Also updated CHANGELOG.md with the latest information.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>
